etna/ciim/client.py

```diff
@@ -1,85 +1,266 @@
+import enum
 import logging
 
+from typing import Optional
+
 import requests
 
-from .exceptions import ConnectionError, InvalidResponse, KongError, KubernetesError
+from .exceptions import ConnectionError, KongError
 
 logger = logging.getLogger(__name__)
 
 
+class Stream(str, enum.Enum):
+    """Options for restricting /search results to a given stream.
+
+    Evidential:
+        Catalogue data i.e: Records
+
+    interpretive:
+        Any content written _about_ evidential i.e: Research Guides, Blog posts
+    """
+
+    EVIDENTIAL = "evidential"
+    INTERPRETIVE = "interpretive"
+
+
+class Sort(str, enum.Enum):
+    """Options for sorting /search results by a given field."""
```

Comment:
Maybe SortBy?, for example, SortBy.TITLE

Reply:
Yeah, I agree `SortBy` is a better name. I'm in two minds though because I'd like the client's method signature to match Kongs spec where possible (the exception being reserved works like `from`):

`        
def search(
    …
    sort: Optional[Sort] = None,
    …
`

Using a `SortBy` enum for a `sort` parameter seems like a mismatch that could cause confusion.

On the other hand, we could name our enums, `<kong-param>Option` i.e: `SortOption` to avoid the issue.

@ababic do you have any thoughts?

Reply:
I prefer `SortBy` too. On the consistency front, I would say that it probably doesn't matter too much. After all, the parameters passed to the API are already in a different case much of the time. I think we should go with `sort_by` / `SortBy` if it makes the client easier to interact with. As long as this it passed to the API as `sort`, we should be good :)

Reply:
Thanks both! Updated to `sort`/`SortBy`